### PR TITLE
gfx: fix vulkan validation errors.

### PR DESCRIPTION
**Pull request by csyonghe**
- Request all Vulkan 1.2 features if they are supported when creating a device.
- Include VK_BUFFER_USAGE_SHADER_BINDING_TABLE_BIT_KHR when creating the buffer for a shader binding table.

### CODE REVIEW

File: tools/gfx/vulkan/vk-device.cpp [hunk 1]
```diff
@@ -124,6 +124,16 @@ Result DeviceImpl::getNativeDeviceHandles(InteropHandles* outHandles)
     return SLANG_OK;
 }
 
+template<typename T>
+static bool _hasAnySetBits(const T& val, size_t offset)
+{
+    uint32_t result = 0;
+    const uint32_t* ptr = reinterpret_cast<const uint32_t*>(&val);
+    for (size_t i = offset; i < sizeof(val) / 4; i++)
```
Comment (expipiplus1) [inline tools/gfx/vulkan/vk-device.cpp:132]: Offset is an offset of 4 byte units...

File: tools/gfx/vulkan/vk-device.cpp [hunk 2]
```diff
@@ -589,6 +561,14 @@ Result DeviceImpl::initVulkanInstanceAndDevice(
             m_features.add("realtime-clock");
         }
 
+        if (_hasAnySetBits(
+                extendedFeatures.vulkan12Features,
+                offsetof(VkPhysicalDeviceVulkan12Features, pNext) + sizeof(void*)))
```
Comment (expipiplus1) [inline tools/gfx/vulkan/vk-device.cpp:566]: ... but offset here is in bytes

**Review by expipiplus1 — APPROVED** [review]

File: tools/gfx/vulkan/vk-device.cpp [hunk 3]
```diff
@@ -124,6 +124,16 @@ Result DeviceImpl::getNativeDeviceHandles(InteropHandles* outHandles)
     return SLANG_OK;
 }
 
+template<typename T>
+static bool _hasAnySetBits(const T& val, size_t offset)
+{
+    uint32_t result = 0;
+    const uint32_t* ptr = reinterpret_cast<const uint32_t*>(&val);
+    for (size_t i = offset; i < sizeof(val) / 4; i++)
+        result |= ptr[i];
```
[the remark this answers is not in the document]
Comment (csyonghe) [reply tools/gfx/vulkan/vk-device.cpp]: Fixed.

File: tools/gfx/vulkan/vk-device.cpp (hunk 1 above)
Comment (csyonghe) [reply tools/gfx/vulkan/vk-device.cpp]: Fixed.

**Pull request merged**